text_formatter.go

```diff
@@ -73,10 +82,12 @@ func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
 	isColorTerminal := isTerminal && (runtime.GOOS != "windows")
 	isColored := (f.ForceColors || isColorTerminal) && !f.DisableColors
 
+	f.Mutex.Lock()
 	timestampFormat := f.TimestampFormat
```

Comment:
I cannot share with you that code, it's a private repo, to reproduce it i've made a simple example:
> cat logRc.go 

package main

import(
	"github.com/Sirupsen/logrus" 	
)

func main() {
	var log = logrus.New()

	for i:=0; i<10; i++ {
		go func(){
			log.Info("foo")
		}()
	}
}


> go run --race logRc.go

INFO[0000] foo                                          
==================
WARNING: DATA RACE
Read by goroutine 7:
  github.com/Sirupsen/logrus.(*TextFormatter).Format()
      /home/marcos/go/src/github.com/Sirupsen/logrus/text_formatter.go:76 +0x787
  github.com/Sirupsen/logrus.(*Entry).Reader()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:41 +0xa1
  github.com/Sirupsen/logrus.(*Entry).log()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:84 +0x351
  github.com/Sirupsen/logrus.(*Entry).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:119 +0xad
  github.com/Sirupsen/logrus.(*Logger).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logger.go:122 +0x15d
  main.main.func1()
      /home/marcos/go/logRc.go:12 +0xfb

Previous write by goroutine 6:
  github.com/Sirupsen/logrus.(*TextFormatter).Format()
      /home/marcos/go/src/github.com/Sirupsen/logrus/text_formatter.go:77 +0x7a7
  github.com/Sirupsen/logrus.(*Entry).Reader()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:41 +0xa1
  github.com/Sirupsen/logrus.(*Entry).log()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:84 +0x351
  github.com/Sirupsen/logrus.(*Entry).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:119 +0xad
  github.com/Sirupsen/logrus.(*Logger).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logger.go:122 +0x15d
  main.main.func1()
      /home/marcos/go/logRc.go:12 +0xfb

Goroutine 7 (running) created at:
  main.main()
      /home/marcos/go/logRc.go:13 +0x217

Goroutine 6 (finished) created at:
  main.main()
      /home/marcos/go/logRc.go:13 +0x217
==================
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
Found 1 data race(s)
exit status 66